services/user/Permissions/plugin/wit/world.wit

```diff
@@ -2,15 +2,15 @@ package permissions:plugin;
 
 interface api {
     use host:common/types.{error};
-    save-permission: func(caller: string, callee: string) -> result<_, error>; // only option in v1 is fullAccess
-}
-
-interface users {
-    use host:common/types.{error};
-    is-permitted: func(caller: string) -> result<bool, error>;
+    /// @description Check if the caller is permitted access to `callee`'s privileged methods
+    /// and if not already, prompt user to request such access.
+    /// @param caller The caller, requesting access
+    is-auth-or-prompt: func(caller: string) -> result<bool, error>;
```

Comment:
Not necessary to address this comment in this PR (I won't hold up the PR for this), 

After a little more thought, I'm thinking maybe a better API would be:
* is-auth 
* prompt

So `is-auth` stays the same, but the other function, instead of `is-auth-or-prompt` just becomes a `prompt` function. The caller, then, is responsible for prompting at will: 
```rust
if permissions.is_auth(...) != Ok(true) {
    permissions.prompt(...)
}
```

This has the added benefit of allowing the caller to `prompt` regardless of the is-auth status (which allows the UI to allow the user to reconfigure their permissions even if already granted).

Thoughts?

Reply:
Well...
In the not-Permissions scenario, the use has Supervisor::prompt_user() it can call directly, right?
For the Permissions scenario, I wouldn't call it just prompt cuz it's not that generic. I suppose prompt_auth()?
Lastly, we actually *don't* have a flow for removing an auth. Shall I add that to the Deny button? I think that's a super noninvasive way to get this done. No change at all to the current functionality on a first-time prompt, but it'll clear the auth grant if it has previously been made. The only corner case that leaves is the "shit, I didn't want to even go here, let me cancel, which of course the user always has the browser back button for.